python_modules/dagster/dagster/_core/definitions/definitions_load_context.py

```diff
@@ -57,13 +58,26 @@ def get(cls) -> "DefinitionsLoadContext":
     @classmethod
     def set(cls, instance: "DefinitionsLoadContext") -> None:
         """Get the current DefinitionsLoadContext."""
+        current_pending_metadata = (
+            cls._instance.get_pending_reconstruction_metadata() if cls._instance else {}
+        )
+        instance.replace_pending_reconstruction_metadata(current_pending_metadata)
```

Comment:
hmmm I don't think context should inherit old data by default

Reply:
Yeah, I think the root issue that was leading me to do this was that we call `repository_def_from_target_def` within `repository_def_from_pointer`, resetting the load context. Modified that codepath so that we don't unnecessarily overwrite the load context in this case.